### DISCUSSION TITLE
Minor core improvement ideas

### DISCUSSION THREAD

**Discussion by xoconoch** [kepelet/flo#29] [Ideas]
Hey, I'm loving your project and genuinely think it has the potential to be the best iOS Navidrome client. The UI feels fast and smooth (as an iPhone XR daily driver, that is rather rare to see in most apps) and the main page is the one I love the most out of all the options. I see that this is just getting started and naturally missing some core features such as:

1. A more complete queue manager, that allows you to add (whether append or insert) a track to the queue. As well as a swipe gesture which does some configurable action and of course, the ability to modify the queue itself (i.e. make the 5th song in the queue play immediately next)
2. The ability to download playlists.

Right now, I use Amperfy because I think it's _currently_ the most complete subsonic iOS client.  However, I love the design language you're following and specially, that Flo doesn't remind my poor phone about its years of service (on Amperfy, I cannot scroll through a playlist without it freezing for 5 seconds right after). There are of course a variety of features that could be implemented, but those I just listed would be enough for me to make the switch to Flo. Keep the great work!

**Comment by faultables** [comment]
hi thanks! the ability to download playlists is tracked here #28 and yes I agree with the "real" queue manager. currently it's still a mess because everything is tied tightly to albums (my bad)

i used to use an iPhone XS before developing flo (and was still using an alternative at the time) and i guess we have the same itch

**Comment by criticaldistance** [comment]
I second the queue management need! Enjoying this app so far, but not being able to add an album to the queue is maddening. 

Even something like an “Add to Queue” button next to “Play” and “Shuffle” would be a welcome addition. Or being able to long press an album in the library and add to queue…
